src/Workspaces/Core/Portable/Log/AbstractLogAggregator.cs

```diff
@@ -33,18 +40,18 @@ protected AbstractLogAggregator()
         public void Clear() => _map.Clear();
 
         public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
-            => _map.GetEnumerator();
+            => _map.Select(static kvp => new KeyValuePair<TKey, TValue>((TKey)kvp.Key, kvp.Value)).GetEnumerator();
 
         IEnumerator IEnumerable.GetEnumerator()
             => this.GetEnumerator();
 
         [PerformanceSensitive("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1279909", AllowCaptures = false)]
         protected TValue GetCounter(TKey key)
-            => _map.GetOrAdd(key, _createCounter);
+            => _map.GetOrAdd((object)key, _createCounter);
```

Comment:
⚠️ This is going to box _every time_ a lookup is performed. If this is called heavily, we're moving from a one-time startup cost to a per-call cost.

Reply:
Wouldnt the pay out still be tied to how agressively GC kicks the boxed object out of the heap instead of on every Get for instance?

Reply:
I'm especially worried because the `PerformanceSensitive` attribute indicates that a prior delegate allocation in this method _was_ problematic.

Reply:
@sharwell: lots of callers are things like once-per-completion-session, where this would be immeasurably small.

Reply:
I'm mainly worried about the caller that led to the PerformanceSensitive attribute because the single delegate allocation was adding up.

Reply:
@arkalyanms No, the boxed values are not reused (to my knowledge)

Reply:
While I maintain a belief that the move from O(1) to O(n) for this operation is not the ideal solution, I reviewed the trace that led to the introduction of the specific `[PerformanceSensitive]` attribute and agree with an offline comment by @jasonmalinowski that we can remove the use of this counter from `IncrementalAnalyzerProcessor.Enqueue`, and at that point we have no direct evidence that the new allocation will cause a problem.